lib/beaker/host/unix.rb

```diff
@@ -138,6 +138,23 @@ def external_copy_base
       @external_copy_base
     end
 
+    # Tells you whether a host platform supports beaker's
+    #   {Beaker::HostPrebuiltSteps#set_env} method
+    #
+    # @return [String,nil] Reason message if set_env is not supported,
+    #   nil if it is supported
+    def supports_set_env
```

Comment:
I do agree that it's odd, but I think it's actually the correct way, in this case.

If the method were to be named according to its return value's truthiness, then it would have to be negative.  Negatively named methods are more complex, because the two situations you end up with are these:

1. not something (the method name)
2. not not something (negating the method name)

whereas having just a postively named method ends up having these scenarios:

1. something (method name)
2. not something (negating)

I would also contend that the fact that Ruby has a specific convention for boolean return methods (ending with a question mark) means that my method name implies that it shouldn't be used as a boolean return, and is documented appropriately to match.

If I were to take the altogether different approach of having two return values, one for the actual boolean test and the other being the failure message, then I still wouldn't be able to add a question mark to the end, because it would always return an array (I believe that's what the return value would be boolean resolved to since it's usually implicitly destructured), which would always resolve to true.

@rick, it seems like the most OOP way to resolve this issue would be to have a return object that holds two values, the test value (boolean) and message (string).  The object itself's truthiness would resolve to the test value, and have an accessor to get the message.  However, this seems like a serious amount of overhead for something this small.  What do you think?